cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -1131,11 +1131,13 @@ struct nan_equal_physical_equality_comparator {
  * returns false, representing unequal rows. If the rows are compared without mismatched elements,
  * the rows are equal.
  *
+ * @tparam has_nested_columns compile-time optimization for primitive types
```

Comment:
What does it mean if it is true? I see many places here like
```
has_nested_columns and cudf::is_nested<Element>()
```
Are these not both either always true or both always false?
If not, can the documentation here describe when they do not match?

Reply:
Is this documentation sufficient? If yes, I will copy it where necessary:
```
 * @tparam has_nested_columns compile-time optimization for primitive types.
 *         This template parameter is to be used by the developer by querying
 *         `cudf::detail::has_nested_columns(input)`. `true` compiles operator
 *          overloads for nested types, while `false` only compiles operator
 *          overloads for primitive types.
 * @tparam Nullate A cudf::nullate type describing whether to check for nulls.
 * @tparam PhysicalEqualityComparator A equality comparator functor that compares individual values
 * rather than logical elements, defaults to a comparator for which `NaN == NaN`.
 *
 * NOTE: The operator overloads in sub-class `element_comparator` are templated via the
 *       `type_dispatcher` to help select an overload instance. So, `cudf::is_nested<Element>`
 *       will always return `true`, but it will be a runtime error if template parameter
 *       `has_nested_columns != true`.
 */
```

Reply:
Sorry, I'm still not following this. Will `has_nested_columns == cudf::is_nested<Element>` always be true?
Both of these are compile-time checks. I'm wondering if we need both now.

Reply:
No, it will not. Imagine you instantiate a "equality_comparator.equal_to<false>(view)" where view is a "table_view" with columns that has nested types. There's another operator overload that throws a runtime error for this combination. I believe the last sentence of my NOTE covers it, but I can try to explain it better if it's not clear

Reply:
No this is good. I'm a bit concerned how doxygen will format the `NOTE`. 
Can you move it above the `@tparams` and mark it with a `@note`?

Reply:
Done! Thank you.